components/mediation/inbound-endpoints/org.wso2.micro.integrator.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/jms/JMSUtils.java

```diff
@@ -317,6 +318,7 @@ protected static void handleException(String s) throws NamingException {
                     map.put(headerName, headerValue);
                 }
             }
+            map.remove(BaseConstants.TRANSACTION);
```

Comment:
Why are we removing this in the case of an exception? Shall we have a small comment explaining it.

Reply:
Done.